### PR TITLE
Add pyqg recipe

### PR DESCRIPTION
**Pull request by rabernat**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rabernat** [comment]
I am stuck on this recipe. It can't install because of this error

```
        from Cython.Build import cythonize
    ModuleNotFoundError: No module named 'Cython'
```

File: recipes/pyqg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: 719d7634da033b675431983dde1615f2af702ecf62f61e118584692cfd70210d
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - pyfftw
+    - cython
```
Comment (rabernat) [inline recipes/pyqg/meta.yaml:19]: ...but cython is definitely in the build environment!

Comment (ocefpaf) [reply recipes/pyqg/meta.yaml:19]: It should be in host. I know, not really the best choice of names there.

File: recipes/pyqg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+{% set name = "pyqg" %}
+{% set version = "0.3.0" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pyqg/pyqg/archive/v{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipes/pyqg/meta.yaml:10]: There are a few issues with the tarball. I'll send a PR later to fix them but for now let's use the GH source b/c it has all the files we need.

File: recipes/pyqg/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  run:
+    - python
+    - pyfftw
+    - {{ pin_compatible('numpy') }}
+
+test:
+  imports:
+    - pyqg
+    - pyqg.kernel
+  # Cannot run pip check there is a bogus dependency on cython at runtime.s
```
Comment (ocefpaf) [inline recipes/pyqg/meta.yaml:35]: The package is not pep 517/518 compliant. I'll send a PR later to fix that too.

**Pull request merged**